### Hypothesis Testing Case Study

card usage has been improved sinificantly from last year usage which is 50.

In [7]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
%matplotlib inline
import math

In [2]:
pwd()

u'C:\\Users\\hp\\Downloads\\python\\Stats - 2 [Effect Size]\\Stats - 2 [Effect Size]'

In [6]:
cust=pd.read_csv("cust_seg.csv")

In [4]:
cust.head(5)

custid  sex  AqChannel  region  Marital_status  segment  pre_usage  \
0      70    0          4       1               1        1         57   
1     121    1          4       2               1        3         68   
2      86    0          4       3               1        1         44   
3     141    0          4       3               1        3         63   
4     172    0          4       2               1        2         47   

   Post_usage_1month  Latest_mon_usage  post_usage_2ndmonth  
0                 52              49.2                 57.2  
1                 59              63.6                 64.9  
2                 33              64.8                 36.3  
3                 44              56.4                 48.4  
4                 52              68.4                 57.2

In [5]:
cust.columns

Index([u'custid', u'sex', u'AqChannel', u'region', u'Marital_status',
       u'segment', u'pre_usage', u'Post_usage_1month', u'Latest_mon_usage',
       u'post_usage_2ndmonth'],
      dtype='object')

In [6]:
cust.Latest_mon_usage.mean()

63.17400000000001

In [7]:
cust.Latest_mon_usage.std()

11.242137352892753

### One Sample T-Test


###### card usage has been improved sinificantly from last year usage which is 50.

In [8]:
stats.ttest_1samp(a= cust.Latest_mon_usage,               # Sample data
                 popmean= 50)  # Pop mean

Ttest_1sampResult(statistic=16.57233752433133, pvalue=2.4963719280931583e-39)

This means card usgae has not been imporoved from last year usage which is 50

In [7]:
cust.Latest_mon_usage.mean()

63.17400000000001

### Two Sample T-Test (Paired)

The last campaign was successful in terms of credit card.

In [9]:
print cust.pre_usage.mean()
print cust.post_usage_2ndmonth.mean()

52.23
58.0525


In [10]:
#stats.ttest_rel(a = before, b = after)
stats.ttest_rel(a=cust.pre_usage,
                b=cust.post_usage_2ndmonth)    # Assume samples have equal variance?

Ttest_relResult(statistic=-8.866832246938742, pvalue=4.295733828012836e-16)

The last campaign was successfull in terms of credit card usage

### Two sample T-Test (Independent)

Is there any difference between males and females in terms of credit card usgae?

In [11]:
Males_spend = cust.Post_usage_1month[cust.sex==0]
FeMales_spend = cust.Post_usage_1month[cust.sex==1]

In [12]:
print FeMales_spend.head(3)
print Males_spend.head(3)

1     59
92    62
93    44
Name: Post_usage_1month, dtype: int64
0    52
2    33
3    44
Name: Post_usage_1month, dtype: int64


In [13]:
print Males_spend.mean()
print FeMales_spend.mean()

50.1208791209
54.9908256881


In [14]:
print Males_spend.std()
print FeMales_spend.std()

10.3051606973
8.13371516959


In [15]:
stats.ttest_ind(a= Males_spend,
                b= FeMales_spend,
                equal_var=False)    # Assume samples have equal variance?

Ttest_indResult(statistic=-3.6564080478875276, pvalue=0.00034088493594266187)

In [16]:
stats.ttest_ind(a= Males_spend,
                b= FeMales_spend,
                equal_var=True)    # Assume samples have equal variance?

Ttest_indResult(statistic=-3.7340738531536797, pvalue=0.00024625461203549315)

In [17]:
stats.f_oneway(Males_spend, FeMales_spend)

F_onewayResult(statistic=13.94330754080599, pvalue=0.0002462546120354903)

Yes there is difference between male and femalde credit card usage

### ANOVA

Is there any difference between segments of customers in terms of credit card usage?

In [ ]:
cust.segment.value_counts()

In [19]:
s1 = cust.Latest_mon_usage[cust.segment==1]
s2 = cust.Latest_mon_usage[cust.segment==2]
s3 = cust.Latest_mon_usage[cust.segment==3]

# Perform the ANOVA
stats.f_oneway(s1, s2, s3)

F_onewayResult(statistic=29.279283801321778, pvalue=7.3640108335267396e-12)

In [22]:
cust.Latest_mon_usage.mean()
cust.Latest_mon_usage.std()

11.242137352892753

In [20]:
print s1.mean(); print s2.mean(); print s3.mean()

60.0266666667
68.08
55.704


Yes ,there is difference between segments of customers in terms of credit card usage.

### Chi-squre Test

##### Is there any relation between region and segment?

In [19]:
t=pd.crosstab(cust.segment, cust.region, margins = True)
t

region    1   2   3  All
segment                 
1        16  20   9   45
2        19  44  42  105
3        12  31   7   50
All      47  95  58  200

In [8]:
chisq_df = pd.DataFrame()
cross_tab = pd.crosstab(cust['segment'],cust['region'], margins=False)
stats.chi2_contingency(observed=cross_tab)[1]
temp = pd.DataFrame([stats.chi2_contingency(observed=cross_tab)[0],stats.chi2_contingency(observed=cross_tab)[1] ]).T
temp.columns = ['ChiSquare','P-Value']
chisq_df = pd.concat([chisq_df, temp], axis=0, ignore_index=True)
    
print(chisq_df, '\n')


(   ChiSquare   P-Value
0  16.604442  0.002307, '\n')


Since p value is quite low we can reject the null hypothesis that they are independent and can say that region and sement are related.

### Correlation

Is the relationship between card usage in the latest month and pre usage of campaign?

In [22]:
print np.corrcoef(cust.Latest_mon_usage, cust.pre_usage)
print stats.stats.pearsonr(cust.Latest_mon_usage, cust.pre_usage)

[[1.         0.66228013]
 [0.66228013 1.        ]]
(0.6622801251558604, 1.2767419295068468e-26)


### Simple regression

In [24]:
cust.corr()

custid       sex  AqChannel    region  Marital_status  \
custid               1.000000 -0.087217   0.748793  0.210912        0.482383   
sex                 -0.087217  1.000000   0.001259 -0.125009        0.015337   
AqChannel            0.748793  0.001259   1.000000  0.195364        0.108386   
region               0.210912 -0.125009   0.195364  1.000000        0.136673   
Marital_status       0.482383  0.015337   0.108386  0.136673        1.000000   
segment             -0.058029  0.004009  -0.050060  0.017333       -0.095050   
pre_usage            0.148620 -0.053084   0.243400  0.293322        0.086201   
Post_usage_1month    0.186688  0.256492   0.219999  0.207494        0.127228   
Latest_mon_usage     0.219234 -0.029338   0.202553  0.272459        0.098309   
post_usage_2ndmonth  0.186688  0.256492   0.219999  0.207494        0.127228   

                      segment  pre_usage  Post_usage_1month  Latest_mon_usage  \
custid              -0.058029   0.148620           0.186688          0.219234   
sex                  0.004009  -0.053084           0.256492         -0.029338   
AqChannel           -0.050060   0.243400           0.219999          0.202553   
region               0.017333   0.293322           0.207494          0.272459   
Marital_status      -0.095050   0.086201           0.127228          0.098309   
segment              1.000000  -0.134973          -0.181107         -0.150104   
pre_usage           -0.134973   1.000000           0.596776          0.662280   
Post_usage_1month   -0.181107   0.596776           1.000000          0.617449   
Latest_mon_usage    -0.150104   0.662280           0.617449          1.000000   
post_usage_2ndmonth -0.181107   0.596776           1.000000          0.617449   

                     post_usage_2ndmonth  
custid                          0.186688  
sex                             0.256492  
AqChannel                       0.219999  
region                          0.207494  
Marital_status                  0.127228  
segment                        -0.181107  
pre_usage                       0.596776  
Post_usage_1month               1.000000  
Latest_mon_usage                0.617449  
post_usage_2ndmonth             1.000000

In [ ]:
sm.OLS?

In [ ]:
 statsmodels.tools.add_constant?

In [23]:
import statsmodels.api as sm
Y = cust.Latest_mon_usage
X = sm.add_constant(cust.pre_usage )
mod = sm.OLS(Y, X)
res = mod.fit()
print res.summary()

                            OLS Regression Results                            
Dep. Variable:       Latest_mon_usage   R-squared:                       0.439
Model:                            OLS   Adj. R-squared:                  0.436
Method:                 Least Squares   F-statistic:                     154.7
Date:                Mon, 12 Nov 2018   Prob (F-statistic):           1.28e-26
Time:                        19:20:41   Log-Likelihood:                -709.49
No. Observations:                 200   AIC:                             1423.
Df Residuals:                     198   BIC:                             1430.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         25.2458      3.107      8.125      0.0

Since pearson correlation is 0.66 ,card usage in the latest month and pre usage of campaign are somewhat .correlated